protocol/protocol.go

```diff
@@ -611,18 +634,54 @@ func VerifyReply(replyBytes, publicKey []byte, nonce [NonceSize]byte) (time uint
 	for len(path) > 0 {
 		pathElementIsRight := index&1 == 0
 		if pathElementIsRight {
-			hashNode(&hash, hash[:], path[:sha512.Size])
+			hashNode(&hash, hash[:32], path[:32])
 		} else {
-			hashNode(&hash, path[:sha512.Size], hash[:])
+			hashNode(&hash, path[:31], hash[:32])
 		}
 
 		index >>= 1
-		path = path[sha512.Size:]
+		path = path[31:]
```

Comment:
Why 31, shouldn't they both be 32? Again, see https://tools.ietf.org/html/draft-ietf-ntp-roughtime-01#section-6.3 and https://tools.ietf.org/html/draft-ietf-ntp-roughtime-01#page-11

PS. I was very surprised to see `out[:0]` being used in `hashNode`, but is apparently the case because `Sum(b)` appends the current hash to `b`. And while the resulting size is 0, as long as the capacity is large enough, this works (https://golang.org/ref/spec#Slice_expressions). `Sum` must be the worst interface in Go stdlib.
```go
func hashNode(out *[sha512.Size]byte, left, right []byte) {
	h := sha512.New()
	...
	h.Sum(out[:0])
}